packages/amplify-cli-core/src/state-manager/stateManager.ts

```diff
@@ -39,12 +46,55 @@ export class StateManager {
     return data;
   };
 
+  getDeploymentSecrets = (): $TSAny => {
+    return (
+      JSONUtilities.readJson<$TSAny>(pathManager.getDeploymentSecrets(), {
+        throwIfNotExist: false,
+      }) || {}
+    );
+  };
+
   getProjectTags = (projectPath?: string): Tag[] => ReadValidateTags(pathManager.getTagFilePath(projectPath));
 
   getCurrentProjectTags = (projectPath?: string): Tag[] => ReadValidateTags(pathManager.getCurrentTagFilePath(projectPath));
 
   teamProviderInfoExists = (projectPath?: string): boolean => fs.existsSync(pathManager.getTeamProviderInfoFilePath(projectPath));
 
+  teamProviderInfoHasAuthSecrets = (projectPath?: string): any => {
+    if (this.teamProviderInfoExists(projectPath)) {
+      const teamProviderInfo = this.getTeamProviderInfo(projectPath);
+      const { envName } = this.getLocalEnvInfo();
+      const envTeamProvider = teamProviderInfo[envName];
+      if (envTeamProvider && envTeamProvider.categories && envTeamProvider.categories.auth) {
+        return _.some(Object.keys(envTeamProvider.categories.auth), resource => {
+          return envTeamProvider.categories.auth[resource][hostedUIProviderCredsField];
+        });
+      }
+    }
+    return false;
+  };
+
+  moveSecretsFromTeamProviderToDeployment = (projectPath?: string): void => {
+    const { envName } = this.getLocalEnvInfo(projectPath);
+    let teamProviderInfo = this.getTeamProviderInfo();
+    const envTeamProvider = teamProviderInfo[envName];
+    const amplifyAppId = envTeamProvider.awscloudformation.AmplifyAppId;
+    let secrets = {};
+    Object.keys(envTeamProvider.categories).forEach(category => {
+      if (category === 'auth') {
+        Object.keys(envTeamProvider.categories.auth).forEach(resourceName => {
+          if (envTeamProvider.categories.auth[resourceName][hostedUIProviderCredsField]) {
```

Comment:
same here, filter before the forEach. Also, if only one match is expected (which I think is the case) you can use `.find()` instead of `.filter()`. Can also use find above as there should only be one auth key as well